chain/network/src/peer/peer_actor.rs

```diff
@@ -1149,38 +1199,50 @@ impl PeerActor {
                 let from = &conn.peer_info.id;
                 if msg.expect_response() {
                     tracing::trace!(target: "network", route_back = ?msg.clone(), "Received peer message that requires response");
-                    self.network_state.graph.routing_table.add_route_back(
-                        &self.clock,
-                        msg.hash(),
-                        from.clone(),
-                    );
+                    match conn.tier {
+                        tcp::Tier::T1 => self.network_state.tier1_route_back.lock().insert(
```

Comment:
routeback cache is not very useful in the context of TIER1, as no messages require a response on TIER1 (yet). I've added support for that only for consistency, and it might be removed in the near future.

Reply:
moved to a separate function